app/src/main/kotlin/com/wire/android/ui/home/conversations/model/messagetypes/audio/AudioMessageType.kt

```diff
@@ -0,0 +1,217 @@
+import com.wire.android.media.audiomessage.AudioMediaPlayingState
+import com.wire.android.model.Clickable
+import com.wire.android.ui.common.WireCircularProgressIndicator
+import com.wire.android.ui.common.WireDialog
+import com.wire.android.ui.common.WireDialogButtonProperties
+import com.wire.android.ui.common.WireDialogButtonType
+import com.wire.android.ui.common.button.WireButtonState
+import com.wire.android.ui.common.button.WireSecondaryIconButton
+import com.wire.android.ui.common.clickable
+import com.wire.android.ui.common.colorsScheme
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.spacers.HorizontalSpace
+import com.wire.android.ui.theme.wireColorScheme
+
+@Composable
+fun AudioMessage(
+    audioMediaPlayingState: AudioMediaPlayingState,
+    totalTimeInMs: Int,
+    currentPositionInMs: Int,
+    onPlayButtonClick: () -> Unit,
+    onSliderPositionChange: (Float) -> Unit,
+    onAudioMessageLongClick: (() -> Unit)?
+) {
+    Box(
+        modifier = Modifier
+            .padding(top = dimensions().spacing4x)
+            .background(
+                color = MaterialTheme.wireColorScheme.onPrimary,
+                shape = RoundedCornerShape(dimensions().messageAssetBorderRadius)
+            )
+            .border(
+                width = 1.dp,
+                color = MaterialTheme.wireColorScheme.secondaryButtonDisabledOutline,
+                shape = RoundedCornerShape(dimensions().messageAssetBorderRadius)
+            )
+            .padding(dimensions().spacing8x)
+    ) {
+        if (audioMediaPlayingState is AudioMediaPlayingState.Failed) {
+            FailedAudioMessage()
+        } else {
+            SuccessFullAudioMessage(
+                audioMediaPlayingState = audioMediaPlayingState,
+                totalTimeInMs = totalTimeInMs,
+                currentPositionInMs = currentPositionInMs,
+                onPlayButtonClick = onPlayButtonClick,
+                onSliderPositionChange = onSliderPositionChange,
+                onAudioMessageLongClick = onAudioMessageLongClick
+            )
+        }
+    }
+}
+
+@Composable
+private fun SuccessFullAudioMessage(
+    audioMediaPlayingState: AudioMediaPlayingState,
+    totalTimeInMs: Int,
+    currentPositionInMs: Int,
+    onPlayButtonClick: () -> Unit,
+    onSliderPositionChange: (Float) -> Unit,
+    onAudioMessageLongClick: (() -> Unit)? = null
+) {
+    val audioDuration by remember(currentPositionInMs) {
+        mutableStateOf(
+            AudioDuration(totalTimeInMs, currentPositionInMs)
+        )
+    }
+
+    val isFetching = audioMediaPlayingState is AudioMediaPlayingState.Fetching
+
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(dimensions().audioMessageHeight)
+            .clickable(Clickable(true, onLongClick = onAudioMessageLongClick)),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        WireSecondaryIconButton(
+            minWidth = 32.dp,
```

Comment:
Check designs mocks. I believe size should be 10dp
```suggestion
            dimensions().spacing10x,
```

Reply:
But now looking at the default implementation for `WireSecondaryIconButton` it seems the default size for the inner icon is 16.dp. Maybe let's double check with Wolfgang if design for playing icon should accommodate to this size instead.